google/field_helpers.go

```diff
@@ -0,0 +1,40 @@
+package google
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const networkLinkTemplate = "projects/%s/global/networks/%s"
+
+var networkLinkRegex = regexp.MustCompile(fmt.Sprintf(networkLinkTemplate, "(.+)", "(.+)"))
+
+type NetworkFieldValue struct {
+	Project string
+	Name    string
+}
+
+// Parses a `network` supporting 4 different formats:
+// - https://www.googleapis.com/compute/{version}/projects/myproject/global/networks/my-network
+// - projects/myproject/global/networks/my-network
+// - global/networks/my-network (default project is used)
+// - my-network (default project is used)
+func ParseNetworkFieldValue(network string, config *Config) *NetworkFieldValue {
```

Comment:
Brainstorming outloud here; none of these are action items:

I wonder if there's a way we can eventually merge all the url parsing code; unfortunately it doesn't seem obvious how since each has their own path scheme. Right now the code is relatively short and simple so the cost of having it isn't much.

Maybe at some point we should combine them all into a single namespace? urltools or resourcetools or something?

Reply:
Agree. I think once we have created a few of these FieldValue classes, we will see patterns emerging and we can create common utils to make it easy to create these FieldValue classes.